### DISCUSSION TITLE
[Config Support]:  Moving Media folder to different driver (NOT NAS)

### DISCUSSION THREAD

**Discussion by leonbrag** [blakeblackshear/frigate#12790] [Config Support] [answered]
### Describe the problem you are having

I would like Frigate to store all media in the different hdd on the Debian (highlighted):
![image](https://github.com/user-attachments/assets/a3d1b95a-b03c-47f1-b5b8-ecfcf8ce529e)

I am trying to follow similar steps as this: https://docs.frigate.video/guides/ha_network_storage/

But a bit confused.

1. for this step:
```
database:
  path: /config/frigate.db
```
Do I manually create config directory? If so, where ? 
Under ` /usr/share/hassio/homeassistant` ?

2. Where do I create Frigate's Media folder ?
Do I need to change mount point from` /media/intDrv1` and instead mount it` /media/frigate` ?
Then manually copy content of the `/usr/share/hassio/media/frigate` to `/media/frigate `(which is actually my new separate hdd )?

**OR:**

Do I just create symlink from `/usr/share/hassio/media/frigate` to a new folder that I will create on `/media/intDrv1` ?
Say I create` /media/intDrv1/NewFrigateMedia` then I symlink `/usr/share/hassio/media/frigate` to this newly created location.

Is this a supported method to store Frigate Media in new location ?

### Version

0.13.2-6476f8a

### Frigate config file

```yaml
n/a
```

### Relevant log output

```shell
n/a
```

### Operating system

Debian

### Install method

HassOS Addon

### Object Detector

Coral

### Any other information that may be helpful

Frigate is installed as HASS add-on, Hass is installed on Debian as Supervised
```

CONTAINER ID   IMAGE                                                      COMMAND               CREATED        STATUS                  PORTS                                                                                                                NAMES
888a886786f9   ghcr.io/home-assistant/amd64-hassio-supervisor:latest      "/init"               3 hours ago    Up 3 hours                                                                                                                                   hassio_supervisor
e4ce795a12c1   ghcr.io/blakeblackshear/frigate:0.13.2                     "/init"               8 hours ago    Up 8 hours (healthy)    1935/tcp, 5000/tcp, 8554/tcp, 0.0.0.0:8555->8555/tcp, 0.0.0.0:8555->8555/udp, :::8555->8555/tcp, :::8555->8555/udp   addon_ccab4aaf_frigate-fa
3b01e97bcd9c   ghcr.io/hassio-addons/emqx/amd64:0.6.1                     "/init"               12 hours ago   Up 12 hours (healthy)                                                                                                                        addon_a0d7b954_emqx
1197c4b4f7d1   ccab4aaf/amd64-addon-frigate-proxy:1.5                     "/init"               13 hours ago   Up 13 hours             0.0.0.0:5000->5000/tcp, :::5000->5000/tcp                                                                            addon_ccab4aaf_frigate-proxy
7bcc840d6c25   ghcr.io/home-assistant/qemux86-64-homeassistant:2024.7.4   "/init"               14 hours ago   Up 13 hours                                                                                                                                  homeassistant
2b06ff78be3e   homeassistant/amd64-addon-matter-server:6.4.0              "/init"               14 hours ago   Up 14 hours                                                                                                                                  addon_core_matter_server
39fb8b423795   ghcr.io/home-assistant/amd64-hassio-multicast:2024.03.0    "/init"               36 hours ago   Up 36 hours                                                                                                                                  hassio_multicast
767d5fc61e69   ghcr.io/home-assistant/amd64-hassio-audio:2023.12.0        "/init"               36 hours ago   Up 36 hours                                                                                                                                  hassio_audio
010a5d82a063   ghcr.io/home-assistant/amd64-hassio-dns:2024.04.0          "/init"               36 hours ago   Up 36 hours                                                                                                                                  hassio_dns
56992d5af722   ghcr.io/home-assistant/amd64-hassio-cli:2024.07.0          "/init"               36 hours ago   Up 36 hours                                                                                                                                  hassio_cli
7cd895adea63   ghcr.io/home-assistant/amd64-hassio-observer:2023.06.0     "/usr/bin/observer"   8 days ago     Up 36 hours             0.0.0.0:4357->80/tcp, :::4357->80/tcp                                                                                hassio_observer
```

**Comment by NickM-27** [comment]
There is no option for HA addons to utilize local storage, this feature will have to be added by home assistant. This is one of the reasons that running frigate in docker directly is preferable.

**Comment by hawkeye217** [comment]
Yep, @NickM-27  is right. Since you are running HA supervised on Debian, you'd be better off running the Frigate Proxy addon and just running Frigate via Docker directly, using a docker-compose.yml file. That way you'll be able to map whatever directories you want to use for storage.

**Reply by NickM-27** [reply]
to be clear, when using the ha supervised installation, installing docker containers externally is considered an unsupported install to HA

**Reply by leonbrag** [reply]
Let me make sure I understand what you are saying. The proposal is:

1. remove Frigate addon in Supervised HASS installation
2. install frigate via Docker compose 
3. use Frigate Proxy to integrate Frigate with HASS

But above is NOT supported installation.  Did I get it right ?

If yes, what is a prescribed/supported method to change media location of the Frigate running as add-on HASS Supervised ?

is this via HASS settings  "Change data location" ?

**Reply by NickM-27** [reply]
> If yes, what is a prescribed/supported method to change media location of the Frigate running as add-on HASS Supervised ?

if you are not using NFS then there is no officially supported way to do that

**Reply by NickM-27** [reply]
this is something you would need to ask Home assistant to support

**Reply by ForTheRush** [reply]
> Let me make sure I understand what you are saying. The proposal is:
> 
> 1. remove Frigate addon in Supervised HASS installation
> 2. install frigate via Docker compose 
> 3. use Frigate Proxy to integrate Frigate with HASS
> 
> But above is NOT supported installation.  Did I get it right ?
> 
> If yes, what is a prescribed/supported method to change media location of the Frigate running as add-on HASS Supervised ?
> 
> is this via HASS settings  "Change data location" ?

When I was using home assistant green, my workaround was to create an nfs share on the drive I want to use (even if its attached to the same machine) then i mounted that nfs share to home assistant to use as storage for frigate.  The samba Nas addon was quiet helpful with this. 

In reality once I switched to a proxmox install of HAOS, I just create separate lxc's for stuff like frigate so I can mount whatever storage I feel like to it with ease and use the frigate proxy addon to bring them into HA.  The other benefit of this route is extremely easy snapshots and backups of the system. 

I like to tinker so when I inevitably break something in either HA or frigate, I can just rollback to last nights snapshot and be back up and running again in minutes

**Reply by leonbrag** [reply]
Yes, I understand about NFS (and maybe Samba) to it's own machine. I am sure I don't understand architecture in detail, but not able to set your own media folder running Frigate as HASS add-on (I think add ons are just docker containers) is a very big limitation imo.

**Reply by ForTheRush** [reply]
> but not able to set your own media folder running Frigate as HASS add-on (I think add ons are just docker containers) is a very big limitation imo. 
> 

It's a limitation of Home Assistant. Specifically the supervised version you're running. 
Supervisor will not let you add any other containers to the system and doesn't allow you to edit the docker files for any of the addon's containers directly. 
If you want to be able to edit your containers to specify certain volume paths you should probably switch to a less controlling version of HA.  

>Yes, I understand about NFS (and maybe Samba) to it's own machine. I am sure I don't understand architecture in detail
>

That's not what I just said, I said, as a workaround to use the same machine, you just have to mount an nfs share on the drive. The drive doesn't need to be attached to another machine to do this. Then you use the samba Nas addon (not the regular samba addon) to tell Home Assistant how to map that drive for use with frigate.  

Read the samba Nas docs as they explain how to do exactly what your looking to do. But beyond that don't use a system that controls how you do things and then complain when it does exactly that.

**Reply by leonbrag** [reply]
> If you want to be able to edit your containers to specify certain volume paths you should probably switch to a less controlling version of HA.

Yes, this is exactly what I want! What version should I be using ? I though running HASS Supervised installed on Debian 12 is an ultimate flexibility option.

**Reply by NickM-27** [reply]
Running everything in docker directly provides the most flexibility

**Reply by leonbrag** [reply]
@NickM-27  - is Frigate running in docker container integrated with HASS running in Docker container (using Frigate Integration) a supported scenario ?

**Reply by NickM-27** [reply]
Yes, that is how I run it

**Reply by leonbrag** [reply]
@NickM-27  - thank you for educating me. Curious what makes scenario above supported while integrating Frigate in docker container with HASS Supervised running on Debian not supported?

**Reply by NickM-27** [reply]
because in that scenario the HA supervisor is the one in control of Frigate, and it does not support the feature you want. Meanwhile, when running in docker directly you are in control of frigate and can configure it to work the way that you want

**Reply by ForTheRush** [reply]
> Yes, this is exactly what I want! What version should I be using ? I though running HASS Supervised installed on Debian 12 is an ultimate flexibility option. 
>


The ultimate in flexibility would be running everything separately in their own containers but with having the most flexibility also comes the individual upkeep of many containers vs one home assistant instance. 

Im very happy with the balanced flexibility, performance, and few limitations of my current set up. It wasn't easy getting it to a perfectly stable environment but it's given me all the flexibility I could possibly need.

Currently running a mini PC which had 500gb nvme and 32gb ram that I've since upgraded to a 4tb nvme, 12tb SSD, and 64gb ram. (Little guy is a beast) On this I'm running proxmox as the OS. Proxmox is built on top of Debian 12 right out of the box.  

In proxmox I have home assistant os running in a virtual machine. I have a bunch of lxc's with things like frigate, MQTT, ngnx, tailscale, etc. 
I stuck with HAOS because I liked having the option to keep certain addon's inside HA  while moving others to their own containers.  

Proxmox gives me the ability to use the machine for other things at the same time. I  also have windows virtual machines and have been testing a ton of different Linux distros to find which I like best as a daily driver.  It  takes nightly snapshots of the entire system and can restore back to any point in the past month. It also does week full system backups that I can restore from going back a year. 

Since then I've gotten a couple other machines to use as proxmox nodes for other virtual machines like pihole, adguardhome, vpn, a dns server, and truenas. It's been a fun journey. When I started with home assistant I know absolutely nothing about any of this, I'm not a programmer or dev, or in IT at all (kind of adjacent but not really) and it's became my favorite hobby.

**Reply by leonbrag** [reply]
> Yes, that is how I run it

But Frigate proxy that shows Frigate on HASS sidebar is an add-on. Since Hass in container does not allow add-ons, it will not be possible to have Frigate on side bar, right ? 

That is when Frigate is container and HASS is Container.

**Reply by NickM-27** [reply, marked as answer]
that's not correct, HA allows adding any website as a dashboard in the sidebar
